sdk/core/azure-core-test/src/test_proxy_manager.cpp

```diff
@@ -225,12 +224,50 @@ void TestProxyManager::SetProxySanitizer()
   Azure::Core::Url matcherRequest(m_proxy);
   matcherRequest.AppendPath("Admin");
   matcherRequest.AppendPath("SetMatcher");
-  const std::string matcherBody
-      = "{\"compareBodies\": false ,\"ignoreQueryOrdering\": true , \"ignoredHeaders\": "
-        "\"x-ms-copy-source,x-ms-proposed-lease-id,x-ms-lease-id,x-ms-file-change-"
-        "time,x-ms-file-creation-time,x-ms-file-last-write-time,x-ms-destination-"
-        "lease-id,x-ms-source-lease-id,x-ms-source-content-crc64,x-ms-content-crc64,x-ms-source-"
-        "content-md5,x-ms-content-md5,Content-MD5\"}";
+  std::string matcherBody;
+  {
+    auto jsonRoot = Json::_internal::json::object();
+    jsonRoot["compareBodies"] = false;
+    jsonRoot["ignoreQueryOrdering"] = true;
+    const std::vector<std::string> excludedHeaders = {
+        "Expect",
+        "Connection",
+    };
+    jsonRoot["excludedHeaders"] = std::accumulate(
+        excludedHeaders.begin(),
+        excludedHeaders.end(),
+        std::string(),
+        [](const std::string& lhs, const std::string& rhs) {
+          return lhs + (lhs.empty() ? "" : ",") + rhs;
+        });
+    const std::vector<std::string> ignoredHeaders = {
+        "x-ms-copy-source",
```

Comment:
These headers appear to be specific to storage, would it make sense to have a mechanism which allows each service test to add its own service-specific headers to ignore in recordings?

Otherwise each service which has its own custom headers is going to need to modify the global test base.

Reply:
> would it make sense to have a mechanism which allows each service test to add its own service-specific headers to ignore in recordings?

@LarryOsterman That would be the best. Such mechanism is not the purpose of this PR. (It's not a change that can be done with a few lines of code change. It should be a separate PR.)

I'm not sure how often we will modify these headers or query parameters. So I don't know if the benefit is worth the effort. How about we just leave them here for a while and revisit later?

Reply:
Tracking in https://github.com/Azure/azure-sdk-for-cpp/issues/4372